test/src/graph/nodes/FormatPointsNodeTest.cpp

```diff
@@ -1,39 +1,232 @@
+{
+	rgl_node_t emptyPointCloudOutputNode = nullptr;
+	createOrUpdateNode<EmptyNode>(&emptyPointCloudOutputNode);
+
+	ASSERT_RGL_SUCCESS(rgl_node_points_format(&formatNode, allNotDummyFields.data(), allNotDummyFields.size()));
+	ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(emptyPointCloudOutputNode, formatNode));
+	ASSERT_RGL_SUCCESS(rgl_graph_run(formatNode));
+
+	// When creating outPointCloud, the createFromFormatNode method checks that the outSize matches the field size
+	TestPointCloud outPointCloud = TestPointCloud::createFromFormatNode(formatNode, allNotDummyFields);
+	EXPECT_EQ(outPointCloud.getPointCount(), 0);
+
+	// Check if the contents of outData have changed (they should not have)
+	std::vector<::Field<XYZ_VEC3_F32>::type> outData{
+	    { 2.0f,  3.0f,  5.0f},
+        { 7.0f, 11.0f, 13.0f},
+        {17.0f, 19.0f, 23.0f}
+    };
+	auto outDataCopy = outData;
+
+	// Verify that the data is unchanged; this is a sanity check to make sure that the data is not overwritten
+	ASSERT_RGL_SUCCESS(rgl_graph_get_result_data(formatNode, RGL_FIELD_XYZ_VEC3_F32, outData.data()));
+	ASSERT_EQ(outData.size(), outDataCopy.size());
+	for (int i = 0; i < outData.size(); ++i) {
+		EXPECT_EQ(outData.at(i)[0], outDataCopy.at(i)[0]);
+		EXPECT_EQ(outData.at(i)[1], outDataCopy.at(i)[1]);
+		EXPECT_EQ(outData.at(i)[2], outDataCopy.at(i)[2]);
+	}
+}
+
+TEST_P(FormatPointsNodeTest, should_properly_align_output_size_and_not_change_data)
+{
+	int pointsCount = std::get<0>(GetParam());
+	int numberOfPaddings = std::get<1>(GetParam());
+
+	auto fieldsToFormat = generateRandomStaticFieldsVector(numberOfPaddings);
+	TestPointCloud inPointCloud = TestPointCloud(fieldsToFormat, pointsCount);
+	rgl_node_t inNode = inPointCloud.createUsePointsNode();
+
+	runGraphWithAssertions(inNode, fieldsToFormat);
+
+	TestPointCloud outPointCloud = TestPointCloud::createFromFormatNode(formatNode, fieldsToFormat);
+
+	EXPECT_EQ(outPointCloud, inPointCloud);
+}
+
+TEST_P(FormatPointsNodeTest, multiple_format_should_not_change_data)
+{
+	int pointsCount = std::get<0>(GetParam());
+	int numberOfPaddings = std::get<1>(GetParam());
+
+	auto fieldsToFormat = generateRandomStaticFieldsVector(numberOfPaddings);
+	TestPointCloud pointCloud = TestPointCloud(fieldsToFormat, pointsCount);
+	rgl_node_t inNode = pointCloud.createUsePointsNode();
+
+	std::vector<rgl_node_t> formatNodes(MULTIPLE_FORMATS_COUNT);
+	ASSERT_RGL_SUCCESS(rgl_node_points_format(&formatNodes.at(0), fieldsToFormat.data(), fieldsToFormat.size()));
+	ASSERT_THAT(formatNodes.at(0), testing::NotNull());
+
+	ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(inNode, formatNodes.at(0)));
+	for (int i = 1; i < MULTIPLE_FORMATS_COUNT; ++i) {
+		ASSERT_RGL_SUCCESS(rgl_node_points_format(&formatNodes.at(i), fieldsToFormat.data(), fieldsToFormat.size()));
+		ASSERT_THAT(formatNodes.at(i), testing::NotNull());
+		ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(formatNodes.at(i - 1), formatNodes.at(i)));
+	}
```

Comment:
IMHO format nodes between the first and the last could have different fields.

Reply:
Oh, there's such a test. Do you think there's any benefit of keeping this version?

Reply:
Perhaps it would make more sense to check that **the same** format used repeatedly does not affect the data. But as far as I know RGL does not allow you to create such a graph.